cdap-ui/app/cdap/components/AbstractWidget/SqlConditionsWidget/index.tsx

```diff
@@ -0,0 +1,291 @@
+        .join(', ')}. \n Node names cannot contain "&" "=" "."`;
+      this.setState({
+        error,
+      });
+    }
+    return this.state.error && this.state.error.length;
+  };
+
+  public formatOutput = () => {
+    this.checkRulesForValidStageNames();
+    if (this.state.stageList.length < 2) {
+      this.props.onChange('');
+      return;
+    }
+    const outputArr = [];
+    this.state.rules.forEach((rule) => {
+      const ruleCheck = rule.filter((field) => {
+        return !field.fieldName;
+      });
+      if (ruleCheck.length > 0) {
+        return;
+      }
+      const ruleArr = rule.map((field) => {
+        return field.stageName + '.' + field.fieldName;
+      });
+      outputArr.push(ruleArr.join(' = '));
+    });
+    this.props.onChange(outputArr.join(' & '));
+  };
+
+  public addRule = () => {
+    if (this.state.stageList.length === 0) {
+      return;
+    }
+    const rules = [];
+    this.state.stageList.forEach((stage) => {
+      rules.push({
+        stageName: stage,
+        fieldName: this.state.mapInputSchema[stage][0],
+      });
+    });
+    this.setState({ rules: [...this.state.rules, rules] }, () => {
+      this.formatOutput();
+    });
+  };
+
+  public deleteRule = (index: number) => {
+    const rulesCopy = [...this.state.rules];
+    rulesCopy.splice(index, 1);
+    this.setState({ rules: rulesCopy }, () => {
+      this.formatOutput();
+    });
+  };
+
+  public updateRule = (changedRule: IRule, ruleIdx: number) => {
+    const rules = this.state.rules.map((rule, i) => {
+      if (i === ruleIdx) {
+        return changedRule;
+      }
+      return rule;
+    });
+    this.setState({ rules }, () => {
+      this.formatOutput();
+    });
+  };
+
+  public initializeOptions = () => {
+    return new Promise((resolve, reject) => {
+      const stageList = [];
+      const mapInputSchema = {};
+      this.props.inputSchema.forEach((input) => {
+        stageList.push(input.name);
+        try {
+          mapInputSchema[input.name] = JSON.parse(input.schema).fields.map((field) => {
+            return field.name;
+          });
+        } catch (e) {
+          mapInputSchema[input.name] = [];
+          this.setState(
+            {
+              error: 'Error parsing input schemas.',
+              stageList,
+              mapInputSchema,
+            },
+            () => {
+              reject(e);
+            }
+          );
+        }
+      });
+      if (stageList.length < 2) {
+        this.setState({ error: 'Please connect 2 or more stages.' });
+      }
+      this.setState({ stageList, mapInputSchema }, () => {
+        resolve();
+      });
+    });
+  };
+
+  public init = () => {
+    this.initializeOptions()
+      .then(() => {
+        if (!this.props.value) {
+          this.addRule();
+          return;
+        }
+        const modelSplit = this.props.value.split('&').map((rule) => {
+          return rule.trim();
+        });
+        modelSplit.forEach((rule) => {
```

Comment:
nit: Can we just chain this? Do we need `modelSplit`?

Reply:
@ajainarayanan  keeping this so that it's easier to read/understand. Let me know if you feel otherwise.